registry/storage/garbagecollect.go

```diff
@@ -68,12 +68,14 @@ func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 					return fmt.Errorf("failed to retrieve tags for digest %v: %v", dgst, err)
 				}
 				if len(tags) == 0 {
-					emit("manifest eligible for deletion: %s", dgst)
 					// fetch all tags from repository
 					// all of these tags could contain manifest in history
 					// which means that we need check (and delete) those references when deleting manifest
 					allTags, err := repository.Tags(ctx).All(ctx)
 					if err != nil {
+						if _, ok := err.(distribution.ErrManifestUnknownRevision); !ok {
```

Comment:
hi @thewolt, @squizzi, I have a question about `ErrManifestUnknownRevision`, under what circumstances would this error occur? According to the [code](https://github.com/distribution/distribution/blob/d61d8ebc16770c0b257cb342f12576d59e2df272/registry/storage/tagstore.go#L38), it should return an `ErrRepositoryUnknown` error, please correct me if I have mistaken.

Additionally, I have also submitted a [PR](https://github.com/distribution/distribution/pull/4331) related to this.

Reply:
Sorry but I've never used Go before this PR.
When I brought the test cases from readerx, I found that adding this condition fixed the test and I looked no further.

Using the debugger, it's clear that the error i wanted to catch was indeed `ErrRepositoryUnknown` and the way i've put the condition is useless.

So yes, your PR is correct and provides a more useful message.

Reply:
Thanks for your clarification, I will update it.